### PR TITLE
test: Added test case for requireLong() method at BundleUtils.kt

### PR DESCRIPTION
**Pull request by Rohanraj123**
# Purpose / Description
Improve the test coverage of Ankidroid codebase.

# Fixes
Added test cases for the 'requireLong()' extension function.

# Approach
Added test cases for the `requireLong()` extension function in the `BundleUtils.kt` file.

# Checklist
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
 - [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
 - [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner]

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/BundleUtilsTest.kt [hunk 1]
```diff
@@ -58,6 +60,33 @@ class BundleUtilsTest {
         assertEquals(expected, value)
     }
 
+    @Test
+    fun test_RequireLong_NotFound_ThrowsException() {
+        val mockedBundle = mock(Bundle::class.java)
+
+        whenever(mockedBundle.containsKey(anyString())).thenReturn(false)
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/utils/BundleUtilsTest.kt:67]: I'd have expected we could use strict mock, and ensure the key is only KEY. But I can't easily find how to do it with mockito and its documentation, so fine as is.
Also, I'd trust that if it were easy to do it, and it made sense, @BrayanDSO would have done it in the other test of this class

**Review by Arthur-Milchior — APPROVED** [review]

**Review by lukstbit — COMMENTED** [review]
Thank you for contributing!

I don't think these tests are valuable, maybe just to feel good about higher test coverage. Both tests are mocking all methods used inside and the checks become more about if the if/else clauses are working. A test shouldn't know about how a methods does things inside or what methods it calls.

Ideally the tests for these methods would be implemented as instrumented tests where one can create a bundle with actual state and just call the methods on that bundle to test the behavior/api of the methods and not their inner workings.
I'm fine either way so I'm leaving this to the third reviewer.

**Comment by Rohanraj123** [comment]
> Thank you for contributing!
> 
> I don't think these tests are valuable, maybe just to feel good about higher test coverage. Both tests are mocking all methods used inside and the checks become more about if the if/else clauses are working. A test shouldn't know about how a methods does things inside or what methods it calls.
> 
> Ideally the tests for these methods would be implemented as instrumented tests where one can create a bundle with actual state and just call the methods on that bundle to test the behavior/api of the methods and not their inner workings. I'm fine either way so I'm leaving this to the third reviewer.

Thank you so much for valuable feedback. 
I think test completely follows the basics of unit testing. Because without mocking it is not possible to  test the real class in unit testing. But definately Instrumentation testing would be a good idea to test the functionality of this method. Please let me know.

**Review by david-allison — APPROVED** [review]
Unlikely to pick up a problem, but fine if this was implemented by looking at coverage and finding places where we're lacking.

Would be better to find something a little higher-yield from a testing perspective once you're more familiar with the codebase

Thanks!!!

**Comment by Rohanraj123** [comment]
> Unlikely to pick up a problem, but fine if this was implemented by looking at coverage and finding places where we're lacking.
> 
> Would be better to find something a little higher-yield from a testing perspective once you're more familiar with the codebase
> 
> Thanks!!!

Sure!

**Pull request merged**